src/Sulu/Bundle/AdminBundle/Resources/js/components/Tabs/Tabs.js

```diff
@@ -1,56 +1,398 @@
 // @flow
 import React from 'react';
-import type {ChildrenArray, Element} from 'react';
+import type {ChildrenArray, Element, ElementRef} from 'react';
+import {action, computed, observable} from 'mobx';
+import {observer} from 'mobx-react';
+import classNames from 'classnames';
+import debounce from 'debounce';
+import Popover from '../Popover';
+import Icon from '../Icon';
+import type {Skin} from './types';
 import Tab from './Tab';
+import CollapsedTabList from './CollapsedTabList';
+import CollapsedTab from './CollapsedTab';
 import tabsStyles from './tabs.scss';
 
 type Props = {
     children: ChildrenArray<Element<typeof Tab> | false>,
     onSelect: (tabIndex: number) => void,
     selectedIndex: ?number,
+    skin: Skin,
+    small: boolean,
 };
 
-export default class Tabs extends React.PureComponent<Props> {
+const DEBOUNCE_TIME = 200;
+
+@observer
+class Tabs extends React.Component<Props> {
+    static defaultProps = {
+        skin: 'default',
+        small: false,
+    };
+
+    @observable wrapperWidth: number = 0;
+    @observable tabsWrapperWidth: number = 0;
+    @observable tabsWidth: number = 0;
+
+    @observable tabWidths: Map<number, number> = new Map();
+    @observable tabRefs: Map<number, ?ElementRef<'li'>> = new Map();
+    @observable dropdownOpen = false;
+    @observable lastSelectedIndex: ?number;
+
     static Tab = Tab;
 
+    resizeObserver: ?ResizeObserver;
+
+    containerRef: ?ElementRef<'div'>;
+    wrapperRef: ?ElementRef<'div'>;
+    tabsWrapperRef: ?ElementRef<'div'>;
+    tabsRef: ?ElementRef<'ul'>;
+
+    componentDidMount() {
+        this.resizeObserver = new ResizeObserver(
+            debounce(this.setDimensions, DEBOUNCE_TIME)
+        );
+
+        if (this.wrapperRef) {
+            this.resizeObserver.observe(this.wrapperRef);
+        }
+
+        if (this.tabsWrapperRef) {
+            this.resizeObserver.observe(this.tabsWrapperRef);
+        }
+
+        if (this.tabsRef) {
+            this.resizeObserver.observe(this.tabsRef);
+        }
```

Comment:
At least 2 of them need to be observed, a wrapper and the tabs list. the wrapper because of the available width and the tabs list to trigger the recalculations of the `tabWidths`, because the tabs list has `position: absolute` and it's width will change if the width of a tab changes

Reply:
So is what I said before true? Because if it is, I think the `ResizeObserver` might be called twice if the width of the entire toolbar changes.